### PR TITLE
Adds all modifications to the model

### PR DESCRIPTION
**Pull request by TheSebastianRoss**

### CODE REVIEW

File: config/scenario.json [hunk 1]
```diff
@@ -1,42 +1,58 @@
...
+			"population": 100,
+			"susceptible": [ 0.25, 0.25, 0.25, 0.25 ],
+			"infected": [ 0, 0, 0, 0 ],
+			"recovered": [ 0, 0, 0, 0 ],
+			"deceased": [ 0, 0, 0, 0 ]
+		},
+		"default_config": {
+			"hoya_age": {
+				"susceptibility": [ 0.55, 0.9, 0.85, 0.9 ],
+				"virulence": [ 0.7, 0.8, 0.9, 0.8 ],
```
Comment (romancardenas) [inline config/scenario.json:17]: So, for computing new infections you multiply susceptibility and virulence?

File: config/scenario.json [hunk 2]
```diff
@@ -1,42 +1,58 @@
...
+				"recovery": [ 0.6, 0.6, 0.6, 0.6 ],
+				"mortality": [ 0.00005, 0.00037, 0.03135, 0.08823 ],
+				"infected_capacity": 0.25,
+				"over_capacity_modifier": 6.67,
```
Comment (romancardenas) [inline config/scenario.json:21]: What are `infected_capacity` and `over_capacity_modifier` used for?

File: config/scenario.json [hunk 3]
```diff
@@ -1,42 +1,58 @@
...
+				"phase_penalties": [ 
+					[ 1, 1, 1, 1 ],
+					[ 0.3, 0.5, 0.5, 0.4 ],
+					[ 0.0, 0.2, 0.2, 0.1 ]
+				],
```
Comment (romancardenas) [inline config/scenario.json:26]: I assume that phase penalties correspond to mobility penalties due to lockdown

File: config/scenario.json [hunk 4]
```diff
@@ -1,42 +1,58 @@
...
+				"phase_duration": 4,
+				"disobedience": [ 0.0, 0.0, 0.0, 0.0 ],
+				"mask_use": [ 1, 1, 1, 1 ],
+				"mask_reduction": 0.3,
+				"mask_adoption": 10.0,
+				"precision": 1000
```
Comment (romancardenas) [inline config/scenario.json:32]: What are `mask_reduction` and `mask_adoption` used for?

File: model/hoya_cell.hpp [hunk 5]
```diff
@@ -213,6 +280,45 @@ class hoya_cell : public grid_cell<T, sir, mc> {
...
+    std::vector<float> new_deaths(sir const &last_state) const {
+        std::vector<float> new_d = std::vector<float>();
+		float total_infected = 0;
+        for(int i = 0; i < n_age_segments(); i++) {
+			total_infected += last_state.infected[i];
+            new_d.push_back(last_state.infected[i] * mortality[i]);
+        }
+		if(total_infected > infected_capacity) {
+			for(int i = 0; i < n_age_segments(); i++) {
+				new_d[i] *= over_capacity_modifier;
```
Comment (romancardenas) [inline model/hoya_cell.hpp:294]: You must check that the number of deaths cannot be greater than the number of infected people (for every age segment)

File: config/scenario.json (hunk 1 above)
Comment (TheSebastianRoss) [reply config/scenario.json:17]: Yes, but that would be the susceptibility of the group *contracting* infections multiplied by the virulence of the group *spreading* the infection for each pair considered in the calculations.

File: config/scenario.json (hunk 2 above)
Comment (TheSebastianRoss) [reply config/scenario.json:21]: infected_capacity is the portion of the population which can be infected before the mortality increases (to account for hospitals filling up).

over_capacity_modifier is the amount that the mortality changes after the infected cap is reached.

File: config/scenario.json (hunk 3 above)
Comment (TheSebastianRoss) [reply config/scenario.json:26]: Yes. Each item in the list of phase penalties is a vector that corresponds to how each group would react to that phase of the quarantine.

File: config/scenario.json (hunk 4 above)
Comment (TheSebastianRoss) [reply config/scenario.json:32]: mask_reduction is how much the spread of the disease is reduced to due to masks.

mask_adoption corresponds to how quickly the population starts wearing masks as the cell gains more infected members.

**Review by romancardenas — APPROVED** [review]
It looks quite nice. Well done!

**Pull request merged**